### ISSUE TITLE
[Bug]: Load collection failed when running test_e2e.py after querynode pod recovered from pod kill

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#16049]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20220314-09b2c46b
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.2.dev5
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Load collection failed when running test_e2e.py after querynode pod recovered from pod kill
```
[2022-03-14 19:44:32 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None], kwargs: {'timeout': 20} (api_request.py:55)
[2022-03-14 19:44:52 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:44:32.915813', 'gRPC error': '2022-03-14 19:44:52.916618'}> (decorators.py:81)
[2022-03-14 19:45:13 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:44:53.917351', 'gRPC error': '2022-03-14 19:45:13.918458'}> (decorators.py:81)
[2022-03-14 19:45:34 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:45:14.919938', 'gRPC error': '2022-03-14 19:45:34.922232'}> (decorators.py:81)
[2022-03-14 19:45:55 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:45:35.923542', 'gRPC error': '2022-03-14 19:45:55.924589'}> (decorators.py:81)
[2022-03-14 19:46:16 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:45:56.925656', 'gRPC error': '2022-03-14 19:46:16.926469'}> (decorators.py:81)
[2022-03-14 19:46:37 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:46:17.927545', 'gRPC error': '2022-03-14 19:46:37.928623'}> (decorators.py:81)
[2022-03-14 19:46:58 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:46:38.929495', 'gRPC error': '2022-03-14 19:46:58.930409'}> (decorators.py:81)
[2022-03-14 19:47:19 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:46:59.931399', 'gRPC error': '2022-03-14 19:47:19.9[325](https://github.com/milvus-io/milvus/runs/5543369540?check_suite_focus=true#step:12:325)86'}> (decorators.py:81)
[2022-03-14 19:47:40 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:47:20.9[339](https://github.com/milvus-io/milvus/runs/5543369540?check_suite_focus=true#step:12:339)87', 'gRPC error': '2022-03-14 19:47:40.935549'}> (decorators.py:81)
[2022-03-14 19:48:01 - ERROR - pymilvus.decorators]: grpc RpcError: [load_collection], <_MultiThreadedRendezvous: StatusCode.DEADLINE_EXCEEDED, Deadline Exceeded>, <Time:{'RPC start': '2022-03-14 19:47:41.936595', 'gRPC error': '2022-03-14 19:48:01.937577'}> (decorators.py:81)
[2022-03-14 19:48:01 - ERROR - ci_test]: Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/decorators.py", line 82, in handler
    raise e
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 664, in load_collection
    response = rf.result()
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1647287281.937125285","description":"Error received from peer ipv4:127.0.0.1:19530","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/milvus/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/home/runner/work/milvus/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 453, in load
    conn.load_collection(self._name, timeout=timeout, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pymilvus/decorators.py", line 52, in handler
    raise BaseException(Status.UNEXPECTED_ERROR, "rpc timeout")
pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=rpc timeout)>
```

### Expected Behavior

all test cases passed after pod kill chaos

### Steps To Reproduce

```markdown
see https://github.com/milvus-io/milvus/runs/5543369540?check_suite_focus=true
```

### Anything else?

failed job: https://github.com/milvus-io/milvus/runs/5543369540?check_suite_focus=true

log: https://github.com/milvus-io/milvus/suites/5655871994/artifacts/185187651

**Comment by zhuwenxing** [comment]
Etcd and standalone pod kill also have the same issue.
see: https://github.com/zhuwenxing/milvus/actions/runs/1982495043

**Comment by zhuwenxing** [comment]
/assign @DragonDriver 

Though the error in this issue is load failed, I think it may be related to the #16048

**Comment by DragonDriver** [comment]
[2022/03/14 18:53:02.186 +00:00] [DEBUG] [collection_replica.go:213] ["Successfully add collection "] [collectionID=0]

**Comment by DragonDriver** [comment]
querynode down will cause querycoord to do load balance. In this case, querycoord will migrate all collections/partitions loaded into old querynode to newer querynode.

querycoord schedule the tasks sequencely, such as load collection, release collection and load balance. So if the load balance task can't be finished, the rest tasks in the queue will block. Here, the load request sent by the client blocked.

As for why load balance task blocked, we don't set meta correctly and querycoord will retry the failed tasks and the retried task can't still be succeeded.

I will fix in https://github.com/milvus-io/milvus/pull/16060.

**Comment by zhuwenxing** [comment]
It still failed for querynode pod kill after #16060 merged.  see https://github.com/milvus-io/milvus/runs/5555468232?check_suite_focus=true

**Comment by zhuwenxing** [comment]
It reproduced Stably
https://github.com/zhuwenxing/milvus/runs/5563656216?check_suite_focus=true

**Comment by DragonDriver** [comment]
> It still failed for querynode pod kill after #16060 merged. see https://github.com/milvus-io/milvus/runs/5555468232?check_suite_focus=true

Apologize. I missed the rescheduled tasks.

I noticed that querynode has been killed many times. So querycoord was always trying the load balance task, if target querynode didn't response, the related task will be rescheduled and I forget to set the meta of rescheduled tasks.

**Comment by DragonDriver** [comment]
I will fix it in #16065 .

**Comment by zhuwenxing** [comment]
Verified and passed in version `a3a29fed `
